shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java

```diff
@@ -180,6 +181,384 @@ public void setTextInputEditingState_doesNotInvokeUpdateEditingState() {
+    assertEquals(expectedDelta.getNewSelectionStart(), delta.getNewSelectionStart());
+    assertEquals(expectedDelta.getNewSelectionEnd(), delta.getNewSelectionEnd());
+    assertEquals(expectedDelta.getNewComposingStart(), delta.getNewComposingStart());
+    assertEquals(expectedDelta.getNewComposingEnd(), delta.getNewComposingEnd());
+
+    verify(textInputChannel, times(1)).updateEditingStateWithDeltas(anyInt(), any());
+
+    inputConnection.beginBatchEdit();
+
+    verify(textInputChannel, times(1)).updateEditingStateWithDeltas(anyInt(), any());
+
+    inputConnection.endBatchEdit();
+
+    assertEquals(
+        0,
+        ((ListenableEditingState) textInputPlugin.getEditable())
+            .getBatchTextEditingDeltas()
+            .size());
+
+    verify(textInputChannel, times(1)).updateEditingStateWithDeltas(anyInt(), any());
+
+    inputConnection.beginBatchEdit();
+
+    verify(textInputChannel, times(1)).updateEditingStateWithDeltas(anyInt(), any());
+
+    // Selection changes so this will trigger an update to the framework through
+    // updateEditingStateWithDeltas after the batch edit has completed and notified all listeners
+    // of the editing state.
+    inputConnection.setSelection(3, 4);
+    assertEquals(Selection.getSelectionStart(textInputPlugin.getEditable()), 3);
+    assertEquals(Selection.getSelectionEnd(textInputPlugin.getEditable()), 4);
+
+    verify(textInputChannel, times(1)).updateEditingStateWithDeltas(anyInt(), any());
+
+    verify(textInputChannel, times(1)).updateEditingStateWithDeltas(anyInt(), any());
+
+    inputConnection.endBatchEdit();
+
+    verify(textInputChannel, times(2)).updateEditingStateWithDeltas(anyInt(), any());
+  }
+
+  @Test
+  public void textEditingDelta_TestDeltaIsCreatedWhenComposingTextSetIsReplacing()
+      throws NullPointerException {
+    // Initialize a general TextInputPlugin.
+    InputMethodSubtype inputMethodSubtype = mock(InputMethodSubtype.class);
+    TestImm testImm =
+        Shadow.extract(
+            RuntimeEnvironment.application.getSystemService(Context.INPUT_METHOD_SERVICE));
+    testImm.setCurrentInputMethodSubtype(inputMethodSubtype);
+    View testView = new View(RuntimeEnvironment.application);
+    EditorInfo outAttrs = new EditorInfo();
+    outAttrs.inputType = InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_FLAG_MULTI_LINE;
+    TextInputChannel textInputChannel = spy(new TextInputChannel(mock(DartExecutor.class)));
+    TextInputPlugin textInputPlugin =
+        new TextInputPlugin(testView, textInputChannel, mock(PlatformViewsController.class));
+    CharSequence newText = "helfo";
+    final TextEditingDelta expectedDelta = new TextEditingDelta(newText, "hello", 0, 5, 5, 5, 0, 5);
+
+    // Change InputTarget to FRAMEWORK_CLIENT.
+    textInputPlugin.setTextInputClient(
+        0,
+        new TextInputChannel.Configuration(
+            false,
+            false,
+            true,
+            true,
+            true, // Enable delta model.
```

Comment:
Nit: What do you think of adding a test where this is false and asserting that there are no deltas created and no calls to updateEditingStateWithDeltas?

Reply:
Added tests